From 46569031ead4dd275b5eecda5a6c3f1fa8311138 Mon Sep 17 00:00:00 2001
From: Rami Potinkara <rami.potinkara@qt.io>
Date: Mon, 22 Sep 2025 20:46:01 +0300
Subject: [PATCH 224/553] SKIP three qRhi cases in Android 16

This patch adds SKIP's for
- threeDimTexture [crash]
- storageBufferRuntimeSizeCompute [fail]
- tst_QRhi::leakedResourceDestroy(Vulkan) [crash]

This patch needs re-work later to find the root
causes and fix them.

Task-number: QTBUG-140189
Pick-to: 6.9 6.8
Change-Id: I953ea16f51c33f4d1ca00b8c4252e5fcee9ac808
Reviewed-by: Ville Voutilainen <ville.voutilainen@qt.io>
(cherry picked from commit ced8727a1e5aa0edbe1ab92ba41a7f5b90b783ad)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 tests/auto/gui/rhi/qrhi/tst_qrhi.cpp | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/tests/auto/gui/rhi/qrhi/tst_qrhi.cpp b/tests/auto/gui/rhi/qrhi/tst_qrhi.cpp
index 4176e9af898..28828159b17 100644
--- a/tests/auto/gui/rhi/qrhi/tst_qrhi.cpp
+++ b/tests/auto/gui/rhi/qrhi/tst_qrhi.cpp
@@ -321,6 +321,7 @@ void tst_QRhi::rhiTestDataWithParam(const char *paramName, QSpan<T, E> paramValu
 bool tst_QRhi::isAndroidOpenGLSwiftShader(QRhi::Implementation impl, const QRhi *rhi)
 {
 #ifdef Q_OS_ANDROID
+    qWarning() << rhi->driverInfo();
     if (impl == QRhi::OpenGLES2 && rhi->driverInfo().deviceName.contains("SwiftShader"))
         return true;
 #else
@@ -5599,6 +5600,12 @@ void tst_QRhi::threeDimTexture()
     QFETCH(QRhiInitParams *, initParams);
 
     QScopedPointer<QRhi> rhi(QRhi::create(impl, initParams));
+
+#ifdef Q_OS_ANDROID
+    if (QNativeInterface::QAndroidApplication::sdkVersion() == 36)
+        QSKIP("Fails and crashes on Android 16 (QTBUG-140189)");
+#endif
+
     if (!rhi)
         QSKIP("QRhi could not be created, skipping testing 3D textures");
 
@@ -6287,6 +6294,11 @@ void tst_QRhi::leakedResourceDestroy()
     QFETCH(QRhi::Implementation, impl);
     QFETCH(QRhiInitParams *, initParams);
 
+#ifdef Q_OS_ANDROID
+    if ((QNativeInterface::QAndroidApplication::sdkVersion() == 36) && (impl == QRhi::Vulkan))
+        QSKIP("Fails and crashes on Android 16 (QTBUG-140189)");
+#endif
+
     QScopedPointer<QRhi> rhi(QRhi::create(impl, initParams));
     if (!rhi)
         QSKIP("QRhi could not be created, skipping");
@@ -7157,6 +7169,12 @@ void tst_QRhi::storageBuffer()
     QFETCH(QRhi::Implementation, impl);
     QFETCH(QRhiInitParams *, initParams);
 
+#ifdef Q_OS_ANDROID
+    if (QNativeInterface::QAndroidApplication::sdkVersion() == 36)
+        QSKIP("pipeline->create()) Fails on Android 16 (QTBUG-140189)");
+#endif
+
+
     // we can't test with Null as there is no compute
     if (impl == QRhi::Null)
         return;
-- 
2.51.2

